aicsimageio/readers/default_reader.py

```diff
@@ -111,6 +112,22 @@ def metadata(self) -> Dict[str, Any]:
 
         return self._metadata
 
+    def get_channel_names(self, scene: int = 0) -> Optional[List[str]]:
+        # Check for channel in dims
+        if Dimensions.Channel in self.dims:
+            channel_index = self.dims.index(Dimensions.Channel)
+            channel_dim_size = self.dask_data.shape[channel_index]
+
+            # RGB vs RGBA vs other
+            if channel_dim_size == 3:
+                return ["Red", "Green", "Blue"]
+            elif channel_dim_size == 4:
+                return ["Red", "Green", "Blue", "Alpha"]
+            else:
+                return [str(i) for i in range(channel_dim_size)]
+
+        return None
```

Comment:
I would invert the logic here. 
```
if Dimensions.Channel not in self.dims:
     return None
# rest of the code as normal
```
It also seems like this should raise an exception if it encounters C not being in dims or return a generic name "channel_0" or some such. I don't follow the logic of returning None?

Reply:
Just to clarify all this:  using AbstractMethod implies there is no default implementation and all derived classes are forced to implement it.  Not using it allows you to have a base class implementation that all classes will inherit and are not required to override.   I don't mind either approach for get_channel_names but I wanted to be explicit about the distinction. 

The other question is:  throw or not?  My opinion is that it should not throw (client code should not crash because of calling that function just because C is not a dim).  My other (less strong) opinion is that if channel is not a dim then logically that implies there is one single channel.

Reply:
I can invert the code for sure. But I actually disagree. I think returning None is the proper way to handle this. I understand the thinking of "C" not in dims means a single dimension but I also feel like the statement "C not in dims" means there are _No_ channel dimensions. The difference between zero and None. :woman_shrugging:

Reply:
Specifically I would be weirded out, if it ever came up from someones code:

```python
channels = reader.get_channel_names()
for i, c in enumerate(channels):
    data = reader.data[i, :]
```

This would break because we are returning a channel name, even though the dimension doesn't exist on the data.

Reply:
This seems like something that would be part of the reader spec one way or the other. 

If we don't return channel names for implicit channels then at the aicsimage level we will have to infer them as we are always returning STCZYX which says there is a C and thus the channel needs a name even if it's a default name. It's a matter of preference as to where we want to tackle this but we need to be consistent.

Reply:
I _am_ currently handling it on the AICSImage function. If the reader returns None we fix it there. I feel pretty strongly though about returning None on the reader if there is no channel dimension because of the above examples.